lib/client/api.go

```diff
@@ -1918,6 +1918,9 @@ func (tc *TeleportClient) Join(ctx context.Context, mode types.SessionParticipan
 		return trace.Wrap(err)
 	}
 
+	if session.GetSessionKind() == types.KubernetesSessionKind {
```

Comment:
1. I think the error message can be improved.
2. I would write this with a switch now

```
switch kind := session.GetSessionKind(); kind {
case types.KubernetesSessionKind:
    return trace.BadParameter(...)
case types.SSHSessionKind:
    // continue
default:
    return trace.BadParameter("session joining is not supported for %v sessions", kind)
}
```

Reply:
Please see update

Reply:
Can't we just fall back to `tsh kube join` automatically?
The only thing required is the sessionID and joining mode

Reply:
let me take a look